homeassistant/components/weatherflow/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for WeatherFlow."""
+from __future__ import annotations
+
+import asyncio
+from asyncio import Future
+from asyncio.exceptions import CancelledError
+from typing import Any
+
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.errors import AddressInUseError, ListenerError
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+async def _async_can_discover_devices() -> bool:
+    """Return if there are devices that can be discovered."""
+    future_event: Future[bool] = asyncio.get_running_loop().create_future()
+
+    @callback
+    def _async_found(_):
+        """Handle a discovered device - only need to do this once so."""
+
+        if not future_event.done():
+            future_event.set_result(True)
+
+    async with WeatherFlowListener() as client, asyncio.timeout(10):
+        try:
+            client.on(EVENT_DEVICE_DISCOVERED, _async_found)
+            await future_event
+        except asyncio.TimeoutError:
+            return False
+        except CancelledError:
```

Comment:
I don't see how this can raise CancelledError

Reply:
![image](https://github.com/home-assistant/core/assets/6491743/5f64dd69-db57-48bf-b089-37a47f26ebaa)

Like this? 

I forcibly set the host to `127.0.0.1` which means we will fail discovery. For some reason it seems like the task is getting cancelled instead of timed-out?

Reply:
Do you have a traceback where it comes from?

No worries on leaving it in if you don't as long as we raise from it